src/main/java/fr/insee/pogues/transforms/visualize/PoguesJSONToPoguesJSONDerefImpl.java

```diff
@@ -0,0 +1,114 @@
+    }
+
+    @Override
+    public void transform(InputStream input, OutputStream output, Map<String, Object> params, String surveyName) throws Exception {
+        if (null == input) {
+            throw new NullPointerException(NULL_INPUT_MESSAGE);
+        }
+        if (null == output) {
+            throw new NullPointerException(NULL_OUTPUT_MESSAGE);
+        }
+        String jsonDeref = transform(input, params, surveyName);
+        output.write(jsonDeref.getBytes(StandardCharsets.UTF_8));
+    }
+
+    @Override
+    public String transform(InputStream input, Map<String, Object> params, String surveyName) throws Exception {
+        if (null == input) {
+            throw new NullPointerException(NULL_INPUT_MESSAGE);
+        }
+        return transform(IOUtils.toString(input, StandardCharsets.UTF_8), params, surveyName);
+    }
+
+    @Override
+    public String transform(String input, Map<String, Object> params, String surveyName) throws Exception {
+        if (null == input) {
+            throw new NullPointerException(NULL_INPUT_MESSAGE);
+        }
+        // TODO: This parameter could be replaced by logical check in back-office
+        // (when Pogues-Model supports "childQuestionnaireRef")
+        if (!(boolean) params.get("needDeref")) {
+            logger.info("No de-referencing needed");
+            return input;
+        }
+        Questionnaire questionnaire = transformAsQuestionnaire(input);
+        return PoguesSerializer.questionnaireJavaToString(questionnaire);
+    }
+
+    public Questionnaire transformAsQuestionnaire(String input) throws Exception {
+        if (null == input) {
+            throw new NullPointerException(NULL_INPUT_MESSAGE);
+        }
+        // Parse Pogues json questionnaire
+        JSONParser parser = new JSONParser();
+        JSONObject jsonQuestionnaire = (JSONObject) parser.parse(input);
+        // Get referenced questionnaire identifiers
+        // TODO: The "childQuestionnaireRef" in the json should be supported by Pogues-Model
+        List<String> references = JSONFunctions.getChildReferencesFromQuestionnaire(jsonQuestionnaire);
+        // Deserialize json into questionnaire object
+        Questionnaire questionnaire = PoguesDeserializer.questionnaireToJavaObject(jsonQuestionnaire);
+        //
+        deReference(references, questionnaire);
+        logger.info("Sequences inserted");
+        //
+        return questionnaire;
+    }
+
+    private void deReference(List<String> references, Questionnaire questionnaire) throws Exception {
+        for (String reference : references) {
+            JSONObject referencedJsonQuestionnaire = questionnairesService.getQuestionnaireByID(reference);
+            if (referencedJsonQuestionnaire == null) {
+                logger.warn(
+                        "Null reference behind reference '{}' in questionnaire '{}'.",
+                        reference, questionnaire.getId());
```

Comment:
no need for else here

Reply:
In the [previous implementation](https://github.com/InseeFr/Pogues-Back-Office/commit/8c51caa2d0cb96d9dfaf9db7eeb22df4cc4ee89c) such case would actually have thrown a NullPointerException, I'll remove the else and replace the log.warn with an exception.